File: backend/src/main/java/fruition/aihistory/domain/OperationType.java
package fruition.aihistory.domain;

/**
 * AI 작업 유형.
 *
 * <p>{@code restore_rebuild}는 별도 유형이 아니라 {@link #restore}의 처리 단계다.
 * 재조립 결과 콜백은 새 작업을 만들지 않고 기존 restore 작업의 rebuilding 단계를 완료시킨다.
 */
public enum OperationType {
    /** 문서 AI 편집. 동기 처리이며 저장과 같은 트랜잭션에서 기록한다. */
    document_edit,
    /** 원문 문서를 Wiki page로 만드는 작업. 비동기이며 콜백으로 결과를 받는다. */
    ingest,
    /** 기존 Wiki를 다듬는 작업. 원문 기여가 아니므로 기여 명단을 만들지 않는다. */
    lint,
    /** 특정 작업의 기여를 걷어내는 복구. */
    restore
}

File: services/frontend/src/entities/operation-log/model/operationType.ts
import type { OperationType } from "./types";

/** 작업 유형 표시 문구. 로그 사이드바 항목과 상세 헤더가 함께 쓴다. */
export const OPERATION_TYPE_LABELS: Record<OperationType, string> = {
  document_edit: "AI 편집 반영",
  ingest: "위키 페이지 생성",
  lint: "Lint",
  restore: "롤백"
};

/** Ingest는 작업 종류 대신 시작 시점의 문서 이름을 제목으로 쓴다. */
export function formatOperationLogTitle(
  item: { operation_type: OperationType; target_display_name?: string | null }
): string {
  return item.operation_type === "ingest" && item.target_display_name?.trim()
    ? item.target_display_name
    : OPERATION_TYPE_LABELS[item.operation_type];
}
